pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/DbLoadedSamlClient.java

```diff
@@ -0,0 +1,369 @@
+package org.pac4j.saml.dbclient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opensaml.core.xml.XMLObject;
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.opensaml.saml.metadata.resolver.ChainingMetadataResolver;
+import org.opensaml.saml.metadata.resolver.MetadataResolver;
+import org.opensaml.saml.saml2.core.Attribute;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.encryption.Decrypter;
+import org.opensaml.xmlsec.config.DefaultSecurityConfigurationBootstrap;
+import org.opensaml.xmlsec.impl.BasicSignatureSigningConfiguration;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.ClientType;
+import org.pac4j.core.client.IndirectClient;
+import org.pac4j.core.client.RedirectAction;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.exception.RequiresHttpAction;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.saml.client.SAML2Client;
+import org.pac4j.saml.client.SAML2ClientConfiguration;
+import org.pac4j.saml.context.SAML2ContextProvider;
+import org.pac4j.saml.context.SAML2MessageContext;
+import org.pac4j.saml.context.SAMLContextProvider;
+import org.pac4j.saml.credentials.SAML2Credentials;
+import org.pac4j.saml.crypto.CredentialProvider;
+import org.pac4j.saml.crypto.DefaultSignatureSigningParametersProvider;
+import org.pac4j.saml.crypto.ExplicitSignatureTrustEngineProvider;
+import org.pac4j.saml.crypto.KeyStoreDecryptionProvider;
+import org.pac4j.saml.crypto.SAML2SignatureTrustEngineProvider;
+import org.pac4j.saml.crypto.SignatureSigningParametersProvider;
+import org.pac4j.saml.dbcrypto.KeyStoreCredentialProvider2;
+import org.pac4j.saml.dbmetadata.SAML2IdentityProviderMetadataResolver2;
+import org.pac4j.saml.metadata.SAML2MetadataResolver;
+import org.pac4j.saml.metadata.SAML2ServiceProviderMetadataResolver;
+import org.pac4j.saml.profile.SAML2Profile;
+import org.pac4j.saml.sso.SAML2ObjectBuilder;
+import org.pac4j.saml.sso.SAML2ProfileHandler;
+import org.pac4j.saml.sso.SAML2ResponseValidator;
+import org.pac4j.saml.sso.impl.SAML2AuthnRequestBuilder;
+import org.pac4j.saml.sso.impl.SAML2DefaultResponseValidator;
+import org.pac4j.saml.sso.impl.SAML2WebSSOMessageReceiver;
+import org.pac4j.saml.sso.impl.SAML2WebSSOMessageSender;
+import org.pac4j.saml.sso.impl.SAML2WebSSOProfileHandler;
+import org.pac4j.saml.transport.Pac4jSAMLResponse;
+import org.pac4j.saml.util.Configuration;
+import org.w3c.dom.Element;
+
+import net.shibboleth.utilities.java.support.component.ComponentInitializationException;
+import net.shibboleth.utilities.java.support.resolver.ResolverException;
+
+
+/**
+ * An alternative to {@link SAML2Client}. It accepts {@link DbLoadedSamlClientConfiguration} instead of {@link SAML2ClientConfiguration}.

```

Comment:
Yes, this was my initial idea but I had to introduce the extra client because of the constructor parameter typed to the existing configuration. I think we can get rid of the extra client after making those changes of the Configuration object you propose.